File: app/src/main/java/com/example/farmer/home/HomePagerAdapter.java
package com.example.farmer.home;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.viewpager2.adapter.FragmentStateAdapter;
import androidx.lifecycle.Lifecycle;

import com.example.farmer.farmerai.FragmentFarmerAi;
import com.example.farmer.home.bottomtab.MainFarmExpenditureFragment;
import com.example.farmer.home.bottomtab.MainFertilizerExpenditureFragment;
import com.example.farmer.home.bottomtab.MainWeatherBroadcastFragment;
import com.example.farmer.userprofile.UserProfileFragment;

public class HomePagerAdapter extends FragmentStateAdapter {

    public HomePagerAdapter(@NonNull FragmentManager fragmentManager, @NonNull Lifecycle lifecycle) {
        super(fragmentManager, lifecycle);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch (position) {
            case 1:
                return new MainFertilizerExpenditureFragment();
            case 2:
                return new MainWeatherBroadcastFragment();
            case 3:
                return new UserProfileFragment();
            default:
                return new MainFarmExpenditureFragment();
        }
    }

    @Override
    public int getItemCount() {
        return 4; // Number of tabs
    }
}

File: app/src/main/java/com/example/farmer/home/HomePagerAdapter.kt
package com.example.farmer.home

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.example.farmer.home.bottomtab.MainFarmExpenditureFragment
import com.example.farmer.home.bottomtab.MainFertilizerExpenditureFragment
import com.example.farmer.home.bottomtab.MainWeatherBroadcastFragment
import com.example.farmer.userprofile.UserProfileFragment

class HomePagerAdapter(fragmentManager: FragmentManager, lifecycle: Lifecycle) :
    FragmentStateAdapter(fragmentManager, lifecycle) {
    override fun createFragment(position: Int): Fragment {
        return when (position) {
            1 -> MainFertilizerExpenditureFragment()
            2 -> MainWeatherBroadcastFragment()
            3 -> UserProfileFragment()
            else -> MainFarmExpenditureFragment()
        }
    }

    override fun getItemCount(): Int {
        return 4 // Number of tabs
    }
}